driver-core/src/test/java/com/datastax/driver/core/policies/LatencyAwarePolicyTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ *      Copyright (C) 2012-2014 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core.policies;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.scassandra.http.client.PrimingRequest;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.scassandra.http.client.PrimingRequest.Result.unavailable;
+import static org.scassandra.http.client.PrimingRequest.queryBuilder;
+
+import com.datastax.driver.core.*;
+import com.datastax.driver.core.exceptions.UnavailableException;
+
+public class LatencyAwarePolicyTest extends ScassandraTestBase {
```

Comment:
I realized that there were no tests for LatencyAwarePolicy, so I started a few. However the class is hard to test because it relies on multiple threads to receive updates and compute scores, so I basically am only testing the parts that I have modified (i.e. excluding certains errors from the hosts's score). The core of the class - the computation of the query plan - is still not tested.